src/worker.rs

```diff
@@ -51,19 +51,22 @@ impl<'a> SidekiqWorker<'a> {
         let mut rng = rand::thread_rng();
         let identity: Vec<u8> = iter::repeat(()).map(|()| rng.sample(distributions::Alphanumeric)).take(9).collect();
 
+        // 1. randomize the queue list (which is filled with duplicates based on priority)
         let mut queues = queues.clone();
-        queues.shuffle(&mut rand::thread_rng());
+        queues.shuffle(&mut rng);
+        // 2. remove duplicates. this ensures that some workers will prefer low-priority queues
+        let queues = queues.into_iter().collect::<HashSet<String>>().into_iter().collect();
```

Comment:
[`Vec#dedup()`](https://doc.rust-lang.org/stable/std/vec/struct.Vec.html#method.dedup) only removes consecutive duplicates, which won't work here since we intentionally randomize the order.

The implementation here is equivalent to this Ruby snippet:
```ruby
[1, 2, 1, 2].to_set.to_a
=> [1, 2]